src/Plugin/resource/Field/ResourceFieldBase.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal|restful\Plugin\resource\Field\ResourceFieldBase.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field;
+
+use Drupal\restful\Exception\ServerConfigurationException;
+use Drupal\restful\Http\Request;
+
+abstract class ResourceFieldBase implements ResourceFieldInterface {
```

Comment:
Maybe we should call it ``ResourceEntityFieldBase`` as it specifically deals with field/ properties that can be wrapped with $wrapper?